src/mono/wasi/Wasi.Build.Tests/WasiLibraryModeTests.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using Wasm.Build.Tests;
+
+#nullable enable
+
+namespace Wasi.Build.Tests;
+
+public class WasiLibraryModeTests : BuildTestBase
+{
+    public WasiLibraryModeTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+        : base(output, buildContext)
+    {
+    }
+
+    [Fact]
+    public void ConsoleBuildLibraryMode()
+    {
+        string config = "Release";
+        string id = $"{config}_{GetRandomId()}";
+        string projectFile = CreateWasmTemplateProject(id, "wasiconsole");
+        string code =
+                """
+                using System;
+                using System.Runtime.InteropServices;
+                public unsafe class Test
+                {
+                    [UnmanagedCallersOnly(EntryPoint = "MyCallback")]
+                    public static int MyCallback()
+                    {
+                        Console.WriteLine("WASM Library MyCallback is called");
+                        return 100;
+                    }
+                }
+                """;
+        string csprojCode = 
+                """
+                <Project Sdk="Microsoft.NET.Sdk.WebAssembly">
+                    <PropertyGroup>
+                        <TargetFramework>net9.0</TargetFramework>
```

Comment:
Setting `TFM`, `RID`, `AllowUnsafeBlocks` and `WasmSingleFileBundle` (if we want to test single file bundling) should be enough